doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,304 @@
+[Async Checkpoint Sampling](#async-checkpoint-sampling),
+if the JIT determines that a method is a long running
+interpreted method that would benefit from DLT, the method is
+queued for both a DLT and regular compilation. On a succesful
+DLT compilation, the VM transitions the thread to execute the
+compiled version of the method.
+
+It is worth noting that a DLT compiled method body can only be
+transitioned to from a specific Bytecode PC. It is for this
+reason that a regular compilation is also requested along with
+the DLT compilation; the former ensures that new invocations
+of the method do not have to run interpreted until it reaches
+the Bytecode PC that permits the transition to the DLT compiled
+method body.
+
+
+## Compilation Queue
+
+The Compilation Queue is a weighted queue of compilation
+entries. The weights represent the priority of a compilation;
+in general, the cheaper a compile is expected to be, the
+higher its priority. The queue is synchronized using the
+[Compilation Monitor](#compilation-monitor).
+
+### Low Priority Queue
+
+The JIT also maintains a Low Priority Queue (LPQ). This is
+used to compile methods that are not immediately needed for
+performance, but would benefit the application in the long
+run. Entries from the LPQ are only picked up by Compilation
+Threads when the main Compilation Queue is empty.
+
+
+## Compilation Threads
+
+Compilations are performed on dedicated threads known as Compilation
+Threads. These threads are created at JVM startup, and are
+suspended until there is work to be done. The exact number of
+threads created depends on the resources available on the machine,
+as well as the size of the Compilation Queue.
+
+In general, the Compilation Threads are synchronized using
+the [Compilation Monitor](#compilation-monitor). However, each
+thread also has a Compilation Thread Monitor. This monitor is
+used to synchronize per thread information, such as its State,
+as well as to suspend and resume as needed.
+
+
+## Compilation Monitor
+
+The Compilation Monitor is a global monitor that is used to
+synchronize many aspects of the compilation infrastructure.
+
+
+## Sampler Thread
+
+The Sampler Thread can be though of as a daemon within the JIT. It
+performs several checks periodically. Most notably, it is responsible
+for triggering sampling via Async Checkpoint Sampling.
+
+### Async Checkpoint Sampling
+
+Async Checkpoints, also known as Safe Points and Yield Points, are
+well defined points during a Java Thread's execution wherein the
+thread yields to the VM for the purpose of doing GC or any other
+task that requires the thread to be stopped. Async Checkpoint Sampling
+piggybacks off this infrastructure to obtain sampling information
+that is used to determine which methods should be compiled and
+what what optimization level.
+
+This is the general procedure:
+
+1. Every 10ms, the Sampler Thread sets a flag in the J9VMThead of all Java Threads.
```

Comment:
"of all Java Threads" --> "of all active Java Threads" (they need to have VM access)